tests/bsim/net/compile.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# Copyright 2023 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+# Compile all the applications needed by the network bsim tests
+
+#set -x #uncomment this line for debugging
+set -ue
+
+: "${BSIM_OUT_PATH:?BSIM_OUT_PATH must be defined}"
+: "${BSIM_COMPONENTS_PATH:?BSIM_COMPONENTS_PATH must be defined}"
+: "${ZEPHYR_BASE:?ZEPHYR_BASE must be set to point to the zephyr root\
+ directory}"
+
+WORK_DIR="${WORK_DIR:-${ZEPHYR_BASE}/bsim_out}"
+BOARD="${BOARD:-nrf52_bsim}"
+BOARD_ROOT="${BOARD_ROOT:-${ZEPHYR_BASE}}"
+
+mkdir -p ${WORK_DIR}
+
+source ${ZEPHYR_BASE}/tests/bsim/compile.source
+
+app=samples/net/sockets/echo_server conf_overlay=overlay-802154.conf compile
+app=tests/bsim/net/sockets/echo_test conf_overlay=overlay-802154.conf compile
+
+#Note: Openthread produces a number of macro redefinition warnings, let's not fail becaus of that
```

Comment:
I've opened a PR to get rid of the warnings: https://github.com/zephyrproject-rtos/zephyr/pull/56025